python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -417,148 +472,158 @@ def load_dg_workspace_file_config(path: Path) -> "DgWorkspaceFileConfig":
 
 def _load_dg_file_config(path: Path) -> DgFileConfig:
```

Comment:
with this behavior change `DgFileConfig` seems increasingly inaccurate no?

`DgFileConfig: TypeAlias = Union[DgWorkspaceFileConfig, DgProjectFileConfig]`

We should be thinking in terms of `DgConfigBlock` or something probably

Reply:
>with this behavior change DgFileConfig seems increasingly inaccurate no?

I don't see how this makes `DgFileConfig` more or less accurate. The "FileConfig` _is_ the config block-- it's an abstraction that wraps the config tree that is either mounted on the root (for `dg.toml`) or on `tool.dg.` (for `pyproject.toml`).

There is a pre-existing user config file `dg.toml` that Ben added a while ago, upstack I will do a light refactor to make that more coherent with this system and there will be a third type, `DgUserFileConfig`, added to this union.